editor/project_manager.cpp

```diff
@@ -2473,6 +2533,91 @@ void ProjectManager::_rename_project() {
 	}
 }
 
+void ProjectManager::_manage_project_tags() {
+	for (int i = 0; i < project_tags->get_child_count(); i++) {
+		project_tags->get_child(i)->queue_free();
+	}
+
+	const ProjectList::Item &item = _project_list->get_selected_projects()[0];
+	current_project_tags = item.tags;
+	for (const String &tag : current_project_tags) {
+		ProjectTag *tag_control = memnew(ProjectTag(tag));
+		project_tags->add_child(tag_control);
+		tag_control->connect_button_to(callable_mp(this, &ProjectManager::_delete_project_tag).bind(tag));
+	}
+
+	tag_manage_dialog->popup_centered(Vector2i(500, 0) * EDSCALE);
+}
+
+void ProjectManager::_add_project_tag(const String &p_tag) {
+	if (current_project_tags.has(p_tag)) {
+		return;
+	}
+	current_project_tags.append(p_tag);
+
+	ProjectTag *tag_control = memnew(ProjectTag(p_tag));
+	project_tags->add_child(tag_control);
+	tag_control->connect_button_to(callable_mp(this, &ProjectManager::_delete_project_tag).bind(p_tag));
+}
+
+void ProjectManager::_delete_project_tag(const String &p_tag) {
+	current_project_tags.erase(p_tag);
+	for (int i = 0; i < project_tags->get_child_count(); i++) {
+		ProjectTag *tag_control = Object::cast_to<ProjectTag>(project_tags->get_child(i));
+		if (tag_control && tag_control->get_tag() == p_tag) {
+			memdelete(tag_control);
+			break;
+		}
+	}
+}
+
+void ProjectManager::_apply_project_tags() {
+	ProjectList::Item &item = _project_list->get_selected_projects().write[0];
+
+	PackedStringArray tags;
+	for (int i = 0; i < project_tags->get_child_count(); i++) {
+		ProjectTag *tag_control = Object::cast_to<ProjectTag>(project_tags->get_child(i));
+		if (tag_control) {
+			tags.append(tag_control->get_tag());
+		}
+	}
+
+	ConfigFile cfg;
+	String project_godot = item.path.path_join("project.godot");
+	Error err = cfg.load(project_godot);
+	if (err != OK) {
+		// set_message(vformat(TTR("Couldn't load project at '%s' (error %d). It may be missing or corrupted."), project_godot, err), MESSAGE_ERROR);
```

Comment:
I realized I forgot to implement this. Is there some dialog I can reuse to show error or I need to add yet another one?

Reply:
Ok I added the error to dialog itself:
![image](https://user-images.githubusercontent.com/2223172/226110098-33c3ebe7-ef2a-45f5-ad00-373486118cb2.png)
It will disappear awkwardly for a moment, because OK closes it, but this error can be pretty much only triggered deliberately, so it doesn't matter how it appears.